<launch>
  <!-- Run the main MoveIt! executable without trajectory execution (we do not have controllers configured by default) -->
  <include file="$(find ur16e_moveit_config)/launch/move_group.launch">
    <arg name="allow_trajectory_execution" value="true"/>
    <arg name="fake_execution" value="false"/>
    <arg name="info" value="true"/>
    <arg name="debug" value="false"/>
    <arg name="pipeline" value="ompl"/>
    <arg name="load_robot_description" value="true"/>
  </include>

  <!-- Run Rviz and load the default config to see the state of the move_group node -->
  <include file="$(find ur16e_moveit_config)/launch/moveit_rviz.launch" if="true">
    <arg name="rviz_config" value="$(find ur16e_moveit_config)/launch/moveit.rviz"/>
    <arg name="debug" value="false"/>
  </include>

  <!-- If database loading was enabled, start mongodb as well -->
  <include file="$(find ur16e_moveit_config)/launch/default_warehouse_db.launch" if="false">
    <arg name="moveit_warehouse_database_path" value="$(find ur16e_moveit_config)/default_warehouse_mongo_db"/>
  </include>

  <include file="$(find yolo_bounding_box_detection)/launch/yolo_detection_server.launch"></include>

  <rosparam command="load" file="$(find bark_slidebot)/config/configs.yaml" />

  <node name="publish_config_poses_to_tf" pkg="bark_slidebot" type="publish_config_poses_to_tf.py" respawn="false" output="screen"></node>
  
  <include file="$(find camera_projections)/launch/camera_projections_server.launch"></include>

  <include file="$(find orient_correction)/launch/orient_correction_server.launch"></include>

  <include file="$(find slidebot_detection)/launch/slidebot_detection.launch"></include>

  <node name="bark_slidebot" pkg="bark_slidebot" type="bark_slidebot.py" respawn="false" output="screen"></node>

</launch>